tests/components/esphomelib/test_config_flow.py

```diff
@@ -0,0 +1,120 @@
+"""Test config flow."""
+from collections import namedtuple
+from unittest.mock import patch
+
+import pytest
+
+from tests.common import mock_coro
+
+MockDeviceInfo = namedtuple("DeviceInfo", ["uses_password", "name"])
+
+
+@pytest.fixture()
+def mock_fetch_device_info():
+    """Mock out the fetch device info method."""
+    with patch(
+            'homeassistant.components.esphomelib.config_flow.'
+            'fetch_device_info') as mock_fetch:
+        yield mock_fetch
+
+
+@pytest.fixture()
+def mock_try_login():
+    """Mock out the try login method."""
+    with patch(
+            'homeassistant.components.esphomelib.config_flow.'
+            'try_login') as mock_try:
+        yield mock_try
+
+
+async def test_user_connection_works(hass, mock_fetch_device_info):
+    """Test we can finish a config flow."""
+    result = await hass.config_entries.flow.async_init(
```

Comment:
Look at the hue config flow tests for how to initiate the flow.

Reply:
This is based on the MQTT config flow tests :P Ok, will do